sourcegraph/sourcegraph
# [SG-44071]: Migrate graphiql's prettify to new API

## Description
Migrate api console toolbars wrapper and click actions to use hooks.

## Success criteria
Migrate prettify to the new API and rewrite the wrapper component into a functional one to make it possible.

## Ref
[SG Issue](https://github.com/sourcegraph/sourcegraph/issues/44071)
[Gitstart Ticket](https://app.gitstart.com/clients/sourcegraph/tickets/SG-44071)

## Demo
https://user-images.githubusercontent.com/45232708/200889014-a41f5fc1-b01c-4412-bcdb-fea0561f1e5f.mov

## Test Plan
1. Go to the API console page.
2. Input some query in the query section
3. Try to format it properly using the format button
4. Try to toggle the history sidebar using the history toolbar button
5. Confirm 3, and 4 works.

## App preview:

- [Web](https://sg-web-contractors-sg-44071.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
